ddl/ddl_api.go

```diff
@@ -1879,8 +1884,14 @@ func (d *ddl) handleAutoIncID(tbInfo *model.TableInfo, schemaID int64) error {
 	// The operation of the minus 1 to make sure that the current value doesn't be used,
 	// the next Alloc operation will get this value.
 	// Its behavior is consistent with MySQL.
-	if err = tb.RebaseAutoID(nil, tbInfo.AutoIncID-1, false); err != nil {
-		return errors.Trace(err)
+	if tp == autoid.RowIDAllocType {
```

Comment:
`RowIDAllocType` is confusing and I suggest to rename it to `AutoIncType`.

Reply:
it's a history problem, there are `AutoIncrementType` indeed. (never used)
We have referred it as `RowIDAllocType` in allocator-process, insert-process...
So if someday, auto_increment has separated from row_id, let's refer it as its own name.

Reply:
It's better to use `AutoIncrementType`.

Reply:
I agree using `AutoIncrementType` but let's skip it in this PR.

Reply:
> It's better to use `AutoIncrementType`.
> `RowIDAllocType` is confusing and I suggest to rename it to `AutoIncType`.

Agreed, but since Allocator.`Get()` is using `tp` here and several codes somewhere else, it may be a big change:

https://github.com/pingcap/tidb/pull/16750/files#diff-edfa54cbe2fdb418f0a77bd0ec4a56f2R2221

 Let's  
keep it here and try to refine it later.

Reply:
ok